framework/Source/GPUImageFramebufferCache.m

```diff
@@ -36,10 +36,13 @@ - (id)init;
     }
     
 #if TARGET_IPHONE_SIMULATOR || TARGET_OS_IPHONE
+    __unsafe_unretained __typeof__ (self) weakSelf = self;
     memoryWarningObserver = [[NSNotificationCenter defaultCenter] addObserverForName:UIApplicationDidReceiveMemoryWarningNotification object:nil queue:nil usingBlock:^(NSNotification *note) {
-        
-        [self purgeAllUnassignedFramebuffers];
-	}];
+        __typeof__ (self) strongSelf = weakSelf;
+        if (strongSelf) {
```

Comment:
This if statement will work as expected if weakSelf is changed from __unsafe_unretained to __weak. As it is now, strongSelf will either be a reference to a valid GPUImageFramebufferCache object or else it will be an invalid reference to a deallocated object which will crash the program. But either way it will never be nil, which seems to be what this if statement is checking for.